test/process_bootstrap_test.cc

```diff
@@ -0,0 +1,1155 @@
+
+namespace Nighthawk {
+namespace {
+
+using ::envoy::config::bootstrap::v3::Bootstrap;
+using ::Envoy::StatusHelpers::StatusIs;
+using ::testing::_;
+using ::testing::Invoke;
+using ::testing::NiceMock;
+using ::testing::Return;
+
+// Parses text into Bootstrap.
+absl::StatusOr<Bootstrap> parseBootstrapFromText(const std::string& bootstrap_text) {
+  Bootstrap bootstrap;
+  if (!Envoy::Protobuf::TextFormat::ParseFromString(bootstrap_text, &bootstrap)) {
+    return absl::InvalidArgumentError(
+        fmt::format("cannot parse bootstrap text:\n{}", bootstrap_text));
+  }
+  return bootstrap;
+}
+
+class CreateBootstrapConfigurationTest : public testing::Test {
+protected:
+  CreateBootstrapConfigurationTest() = default;
+
+  // Resolves all the uris_, so they can be passed to createBootstrapConfiguration().
+  void resolveAllUris() {
+    ON_CALL(mock_dispatcher_, createDnsResolver(_, _)).WillByDefault(Return(mock_resolver_));
+
+    EXPECT_CALL(*mock_resolver_, resolve(_, _, _))
+        .WillRepeatedly(Invoke([](const std::string&, Envoy::Network::DnsLookupFamily,
+                                  // Even though clang-tidy is right, we cannot
+                                  // change the function declaration here.
+                                  // NOLINTNEXTLINE(performance-unnecessary-value-param)
+                                  Envoy::Network::DnsResolver::ResolveCb callback) {
+          callback(Envoy::Network::DnsResolver::ResolutionStatus::Success,
+                   Envoy::TestUtility::makeDnsResponse({"127.0.0.1"}));
+          return nullptr;
+        }));
+
+    for (const UriPtr& uri : uris_) {
+      uri->resolve(mock_dispatcher_, Envoy::Network::DnsLookupFamily::Auto);
+    }
+
+    if (request_source_uri_ != nullptr) {
+      request_source_uri_->resolve(mock_dispatcher_, Envoy::Network::DnsLookupFamily::Auto);
+    }
+  }
+
+  std::shared_ptr<Envoy::Network::MockDnsResolver> mock_resolver_{
+      std::make_shared<Envoy::Network::MockDnsResolver>()};
+  NiceMock<Envoy::Event::MockDispatcher> mock_dispatcher_;
+  std::vector<UriPtr> uris_;
+  UriPtr request_source_uri_;
+  int number_of_workers_{1};
+};
+
+TEST_F(CreateBootstrapConfigurationTest, FailsWithoutUris) {
+  std::unique_ptr<Client::OptionsImpl> options =
+      Client::TestUtility::createOptionsImpl("nighthawk_client https://www.example.org");
+
+  absl::StatusOr<Bootstrap> bootstrap =
+      createBootstrapConfiguration(*options, uris_, request_source_uri_, number_of_workers_);
+  ASSERT_THAT(bootstrap, StatusIs(absl::StatusCode::kInvalidArgument));
+}
+
+TEST_F(CreateBootstrapConfigurationTest, CreatesBootstrapForH1) {
+  uris_.push_back(std::make_unique<UriImpl>("http://www.example.org"));
+  resolveAllUris();
+
+  std::unique_ptr<Client::OptionsImpl> options =
+      Client::TestUtility::createOptionsImpl("nighthawk_client http://www.example.org");
+
+  absl::StatusOr<Bootstrap> expected_bootstrap = parseBootstrapFromText(R"pb(
+    static_resources {
```

Comment:
I would move them to separate files if they were longer. As it stands my motivation was test case readability, since these tests also serve as a documentation of how our bootstrap looks like for various Nighthawk options. When the reader has the generated bootstrap right next to the options, they can see everything on one screen.

Let me know if you don't think this is worth keeping them in.

Reply:
Allright, happy with this as is, that makes sense.